File: codegen/results/v002/gemma4_31b_think/20260414_075045/python/004_json_filter/solution.py
import json

def solve():
    try:
        # Read the input file
        with open('input/data.json', 'r') as f:
            data = json.load(f)
        
        # Filter records: active is true and age is 30 or older
        filtered_data = [
            record for record in data 
            if record.get('active') is True and record.get('age', 0) >= 30
        ]
        
        # Sort records by name ascending
        sorted_data = sorted(filtered_data, key=lambda x: x['name'])
        
        # Output the result as a JSON array to stdout
        print(json.dumps(sorted_data))
        
    except (FileNotFoundError, json.JSONDecodeError):
        # Since the prompt requires no other output, 
        # we handle errors silently or as per standard Python behavior.
        pass

if __name__ == "__main__":
    solve()

File: codegen/results/v002/gemma4_31b_think/20260414_075045/go/004_json_filter/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"sort"
)

// Record represents the structure of the input JSON objects
type Record struct {
	Name   string  `json:"name"`
	Age    int     `json:"age"`
	Active bool    `json:"active"`
	Score  float64 `json:"score"`
}

func main() {
	// Read the input file
	data, err := os.ReadFile("input/data.json")
	if err != nil {
		os.Exit(1)
	}

	var records []Record
	if err := json.Unmarshal(data, &records); err != nil {
		os.Exit(1)
	}

	// Filter records: active is true and age is 30 or older
	var filtered []Record
	for _, r := range records {
		if r.Active && r.Age >= 30 {
			filtered = append(filtered, r)
		}
	}

	// Sort records by name ascending
	sort.Slice(filtered, func(i, j int) bool {
		return filtered[i].Name < filtered[j].Name
	})

	// Marshal the result back to JSON
	output, err := json.Marshal(filtered)
	if err != nil {
		os.Exit(1)
	}

	// Output the JSON array to stdout
	fmt.Println(string(output))
}